### PR TITLE
MON-1798 - Added content for disabling the local Alertmanager in the Monitoring stack

### PR DESCRIPTION
**Pull request by bburt-rh**
- Aligned team: Dev Tools
- For branches: 4.9+
- Jira: https://issues.redhat.com/browse/MON-1798
- Direct link to doc preview: https://deploy-preview-36300--osdocs.netlify.app/openshift-enterprise/latest/monitoring/configuring-the-monitoring-stack.html#disabling_the_local_alertmanager_configuring-the-monitoring-stack
- SME review: @fpetkovski (approved)
- QE review: @juzhao (approved)
- Peer review: @rolfedh (approved)
- All reviews complete. Please merge now.

### CODE REVIEW

**Comment by bburt-rh** [comment]
@rolfedh: don't know why the `All checks have failed message` is now appearing above.

File: modules/monitoring-disabling-the-local-alertmanager.adoc [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+//
+// * monitoring/configuring-the-monitoring-stack.adoc
+
+[id="disabling_the_local_alertmanager_{context}"]
+= Disabling the local Alertmanager
+
+A default installation of the {product-title} monitoring stack includes in the `openshift-monitoring` project an Alertmanager instance that routes alerts from Prometheus instances.
+
+You can disable this Alertmanager instance by configuring the `cluster-monitoring-config` `ConfigMap` object in the `openshift-monitoring` project.
+You might do so in single-node deployments that do not need to deploy a local Alertmanager cluster because alerts are centralized at the core (for example, in the hub cluster) or because running Alertmanager locally would take resources away from user workloads and add management overhead.
```
Comment (rolfedh) [inline modules/monitoring-disabling-the-local-alertmanager.adoc:11]: How might we improve the readability of this phrase?

File: modules/monitoring-disabling-the-local-alertmanager.adoc [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  config.yaml: |
+    alertmanagerMain:
+      enabled: false
+----
+
+. Save the file to apply the changes. The Alertmanager instance is disabled automatically when you apply the change.
++
+[WARNING]
+====
+When changes are saved to the `cluster-monitoring-config` `ConfigMap` object, the pods and other resources in the `openshift-monitoring` project might be redeployed. The running monitoring processes in that project might also be restarted.
```
Comment (rolfedh) [inline modules/monitoring-disabling-the-local-alertmanager.adoc:47]: Can you rephrase this using the active voice? E.g., "Saving changes to the `cluster-monitoring-config` `ConfigMap` object might redeploy the pods and other resources in the `openshift-monitoring` project and restart the running monitoring processes in that project."

File: monitoring/configuring-the-monitoring-stack.adoc [hunk 3]
```diff
@@ -140,6 +140,13 @@ include::modules/monitoring-disabling-grafana.adoc[leveloffset=+1]
 
 * See xref:../monitoring/configuring-the-monitoring-stack.adoc#preparing-to-configure-the-monitoring-stack[Preparing to configure the monitoring stack] for steps to create monitoring config maps
 
+// Disabling the local Alertmanager
+include::modules/monitoring-disabling-the-local-alertmanager.adoc[leveloffset=+1]
+
+.Additional resources
+* See the link:https://prometheus.io/docs/alerting/latest/alertmanager/[Prometheus Alertmanager documentation] for more information about Alertmanager.
+* See xref:../monitoring/managing-alerts.html[Managing alerts] for more information about managing and configuring alerts in the {product-title} monitoring stack.
```
Comment (rolfedh) [inline monitoring/configuring-the-monitoring-stack.adoc:148]: Consider using just the link. Using helper text around the link is no longer required/encouraged.

**Review by rolfedh — COMMENTED** [review]
Hi @bburt-rh. I've left a few suggestions.

File: monitoring/configuring-the-monitoring-stack.adoc (hunk 3 above)
Comment (rolfedh) [reply monitoring/configuring-the-monitoring-stack.adoc:148]:
```suggestion
* See xref:../monitoring/managing-alerts.adoc#managing-alerts_managing-alerts[Managing alerts] for more information about managing and configuring alerts in the {product-title} monitoring stack.
```
It looks like the build is complaining about this xref. I believe we need to reformulate this xref to use the following structure instead. `../monitoring/<filename>.adoc#<bookmark>_<context attribute>

File: modules/monitoring-disabling-the-local-alertmanager.adoc (hunk 1 above)
Comment (bburt-rh) [reply modules/monitoring-disabling-the-local-alertmanager.adoc:11]: @rolfedh Which phrase at which line number?

Comment (rolfedh) [reply modules/monitoring-disabling-the-local-alertmanager.adoc:11]: @bburt-rh Line 11.

File: modules/monitoring-disabling-the-local-alertmanager.adoc [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    alertmanagerMain:
+      enabled: false
+----
+
+. Save the file to apply the changes. The Alertmanager instance is disabled automatically when you apply the change.
++
+[WARNING]
```
Comment (juzhao) [inline modules/monitoring-disabling-the-local-alertmanager.adoc:45]: if we just disable alertmanager, no pods will be restarted

File: modules/monitoring-disabling-the-local-alertmanager.adoc [hunk 5]
```diff
@@ -0,0 +1,49 @@
+// Module included in the following assemblies:
+//
+// * monitoring/configuring-the-monitoring-stack.adoc
+
+[id="disabling_the_local_alertmanager_{context}"]
+= Disabling the local Alertmanager
+
+A default installation of the {product-title} monitoring stack includes in the `openshift-monitoring` project an Alertmanager instance that routes alerts from Prometheus instances.
```
Comment (juzhao) [inline modules/monitoring-disabling-the-local-alertmanager.adoc:8]: change to below is better
`Alertmanager is a default installation of the {product-title} monitoring stack includes in the `openshift-monitoring` project  that routes alerts from Prometheus instances.`

File: modules/monitoring-disabling-the-local-alertmanager.adoc [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+You can disable this Alertmanager instance by configuring the `cluster-monitoring-config` `ConfigMap` object in the `openshift-monitoring` project.
+You might do so in single-node deployments that do not need to deploy a local Alertmanager cluster.
```
Comment (juzhao) [inline modules/monitoring-disabling-the-local-alertmanager.adoc:11]: since we can disable Alertmanager in multi-node cluster. change to below is better
`You might do so if you do not need the local Alertmanager cluster.`

**Review by rolfedh — APPROVED** [review]

File: modules/monitoring-disabling-the-local-alertmanager.adoc (hunk 6 above)
Comment (bburt-rh) [reply modules/monitoring-disabling-the-local-alertmanager.adoc:11]: Ok. That's simpler and more accurate, then.

File: modules/monitoring-disabling-the-local-alertmanager.adoc (hunk 5 above)
Comment (bburt-rh) [reply modules/monitoring-disabling-the-local-alertmanager.adoc:8]: What about this?  

"A local Alertmanager instance that routes alerts from Prometheus instances is enabled by default in the `openshift-monitoring` project of the (product-title} monitoring stack."

File: modules/monitoring-disabling-the-local-alertmanager.adoc (hunk 6 above)
Comment (bburt-rh) [reply modules/monitoring-disabling-the-local-alertmanager.adoc:11]: Combining the two sentences makes it more direct: 

"If you do not need the local Alertmanager instance, you can disable it by configuring the `cluster-monitoring-config` `ConfigMap` object in the `openshift-monitoring` project."

File: modules/monitoring-disabling-the-local-alertmanager.adoc (hunk 4 above)
Comment (bburt-rh) [reply modules/monitoring-disabling-the-local-alertmanager.adoc:45]: Will remove the warning

File: monitoring/configuring-the-monitoring-stack.adoc (hunk 3 above)
Comment (bburt-rh) [reply monitoring/configuring-the-monitoring-stack.adoc]: Fixed. Thanks.

File: modules/monitoring-disabling-the-local-alertmanager.adoc (hunk 1 above)
Comment (bburt-rh) [reply modules/monitoring-disabling-the-local-alertmanager.adoc]: Reworked and simplified the phrasing. Thanks!

**Review by rolfedh — APPROVED** [review]

File: modules/monitoring-disabling-the-local-alertmanager.adoc [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    alertmanagerMain:
+      enabled: false
+----
+
+. Save the file to apply the changes. The Alertmanager instance is disabled automatically when you apply the change.
```
Comment (rolfedh) [inline modules/monitoring-disabling-the-local-alertmanager.adoc:42]: I realize this is a boilerplate phrase borrowed from other related topics, but does saving the file really apply the changes? Or must one use an `oc apply` similar to [the one shown here](https://deploy-preview-36300--osdocs.netlify.app/openshift-enterprise/latest/monitoring/configuring-the-monitoring-stack.html#creating-cluster-monitoring-configmap_configuring-the-monitoring-stack)?

**Review by rolfedh — APPROVED** [review]
I asked a question about the content. Otherwise, this is good to go!

File: modules/monitoring-disabling-the-local-alertmanager.adoc (hunk 7 above)
Comment (bburt-rh) [reply modules/monitoring-disabling-the-local-alertmanager.adoc:42]: I'm not sure. @fpetkovski or @juzhao would know the answer, I'm sure. The phrase (or something similar) appears throughout the configuration section of the Monitoring docs.

Comment (juzhao) [reply modules/monitoring-disabling-the-local-alertmanager.adoc:42]: cluster-monitoring-operator would watch and apply the changes
1. if the `cluster-monitoring-config` configmap does not exist, we can use `oc applly` to create and apply the changes
2. if the `cluster-monitoring-config` configmap exists, edit and save `cluster-monitoring-config` configmap would apply the changes

File: modules/monitoring-disabling-the-local-alertmanager.adoc [hunk 8]
```diff
@@ -0,0 +1,43 @@
+// Module included in the following assemblies:
+//
+// * monitoring/configuring-the-monitoring-stack.adoc
+
+[id="disabling_the_local_alertmanager_{context}"]
```
Comment (abrennan89) [inline modules/monitoring-disabling-the-local-alertmanager.adoc:5]:
```suggestion
[id="monitoring-disabling-the-local-alertmanager_{context}"]
```
This should match the filename exactly: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#module-file-metadata and should use dashes instead of underscores

File: modules/monitoring-disabling-the-local-alertmanager.adoc [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+= Disabling the local Alertmanager
+
+A local Alertmanager that routes alerts from Prometheus instances is enabled by default in the `openshift-monitoring` project of the {product-title} monitoring stack.
+
+If you do not need the local Alertmanager, you can disable it by configuring the `cluster-monitoring-config` `ConfigMap` object in the `openshift-monitoring` project.
+
+.Prerequisites
+
+* You have access to the cluster as a user with the `cluster-admin` role.
+* You have created the `cluster-monitoring-config` `ConfigMap` object.
```
Comment (abrennan89) [inline modules/monitoring-disabling-the-local-alertmanager.adoc:15]:
```suggestion
* You have created the `cluster-monitoring-config` config map.
```
This might be nicer for readability instead of having so many backticks? Just a suggestion though.

File: modules/monitoring-disabling-the-local-alertmanager.adoc [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+* You have installed the OpenShift CLI (`oc`).
+
+.Procedure
+
+. Edit the `cluster-monitoring-config` `ConfigMap` object in the `openshift-monitoring` project:
```
Comment (abrennan89) [inline modules/monitoring-disabling-the-local-alertmanager.adoc:20]: same suggestion as above re using "config map" instead of "`ConfigMap` object" - they mean the same thing, it's up to you though.

**Review by abrennan89 — COMMENTED** [review]
One required change re the heading ID and a couple of optional suggestions before we merge this 🙂

File: modules/monitoring-disabling-the-local-alertmanager.adoc (hunk 10 above)
Comment (bburt-rh) [reply modules/monitoring-disabling-the-local-alertmanager.adoc]: I like this suggestion. Change made.

File: modules/monitoring-disabling-the-local-alertmanager.adoc (hunk 9 above)
Comment (bburt-rh) [reply modules/monitoring-disabling-the-local-alertmanager.adoc]: I like this suggestion. Change made.

File: modules/monitoring-disabling-the-local-alertmanager.adoc (hunk 8 above)
Comment (bburt-rh) [reply modules/monitoring-disabling-the-local-alertmanager.adoc]: Fixed.
